posgram/Service/FriendStService.cs

```
@@ -0,0 +1,47 @@
+﻿using posgram.Models;
+using posgram.Repository.IRepository;
+using posgram.Service.IService;
+
+namespace posgram.Service
+{
+    public class FriendStService : IFriendStService
+    {
+        private readonly IFriendStRepository _friendStRepository;
+        public FriendStService(IFriendStRepository friendStRepository)
+        {
+            _friendStRepository = friendStRepository;
+        }
+
+        public async Task Create(FriendSt entity)
+        {
+            await _friendStRepository.Create(entity);
+        }
+
+        public async Task Delete(long id)
+        {
+            var status = await _friendStRepository.GetById(id);
+            if (status == null)
+            {
+                throw new ArgumentNullException("There is no status with this id");
+            }
+            await _friendStRepository.Delete(status);
+        }
+
+        public async Task<List<FriendSt>> GetAll()
+        {
+            var statuses = await _friendStRepository.GetAll();
+            return statuses;
+        }
+
+        public async Task<FriendSt> GetById(long id)
+        {
+            var status = await _friendStRepository.GetById(id);
+            return status;
+        }
```

What happens if there is no status with such id? Maybe we need to throw an exception if status is null, what do you think?

---

Ok